lib/jekyll-github-metadata.rb

```diff
@@ -61,9 +62,18 @@ def client
         @client ||= Client.new
       end
 
+      def repository
+        @repository ||= GitHubMetadata::Repository.new(nwo).tap do |repo|
+          Jekyll::GitHubMetadata.log :debug, "Generating for #{repo.nwo}"
+        end
+      end
+
+      def nwo
+        @nwo ||= GitHubMetadata::RepositoryFinder.nwo
```

Comment:
If we did that, JekyllGitHubMetadata would have to know (or be passed) the site variable (which you seemed to indicate the opposite below).